istioctl/cmd/proxyconfig.go

```diff
@@ -526,34 +616,45 @@ func proxyConfig() *cobra.Command {
 	endpointConfigCmd.PersistentFlags().IntVar(&port, "port", 0, "Filter endpoints by Port field")
 	endpointConfigCmd.PersistentFlags().StringVar(&clusterName, "cluster", "", "Filter endpoints by cluster name field")
 	endpointConfigCmd.PersistentFlags().StringVar(&status, "status", "", "Filter endpoints by status field")
+	endpointConfigCmd.PersistentFlags().StringVarP(&configDumpFile, "file", "f", "",
+		"Envoy config dump JSON file")
 
 	bootstrapConfigCmd := &cobra.Command{
-		Use:   "bootstrap <pod-name[.namespace]>",
+		Use:   "bootstrap [<pod-name[.namespace]>]",
 		Short: "Retrieves bootstrap configuration for the Envoy in the specified pod",
 		Long:  `Retrieve information about bootstrap configuration for the Envoy instance in the specified pod.`,
 		Example: `  # Retrieve full bootstrap configuration for a given pod from Envoy.
   istioctl proxy-config bootstrap <pod-name[.namespace]>
 `,
 		Aliases: []string{"b"},
 		Args: func(cmd *cobra.Command, args []string) error {
-			if len(args) != 1 {
+			if (len(args) == 1) != (configDumpFile == "") {
 				cmd.Println(cmd.UsageString())
-				return fmt.Errorf("bootstrap requires pod name")
+				return fmt.Errorf("bootstrap requires pod name or --file parameter")
 			}
 			return nil
 		},
 		RunE: func(c *cobra.Command, args []string) error {
-			podName, ns := handlers.InferPodInfo(args[0], handlers.HandleNamespace(namespace, defaultNamespace))
-			configWriter, err := setupConfigdumpEnvoyConfigWriter(podName, ns, c.OutOrStdout())
+			var configWriter *configdump.ConfigWriter
+			var err error
+			if len(args) == 1 {
+				podName, ns := handlers.InferPodInfo(args[0], handlers.HandleNamespace(namespace, defaultNamespace))
+				configWriter, err = setupPodConfigdumpWriter(podName, ns, c.OutOrStdout())
+			} else {
+				configWriter, err = setupFileConfigdumpWriter(configDumpFile, c.OutOrStdout())
+			}
 			if err != nil {
 				return err
 			}
 			return configWriter.PrintBootstrapDump()
 		},
 	}
 
+	bootstrapConfigCmd.PersistentFlags().StringVarP(&configDumpFile, "file", "f", "",
+		"Envoy config dump JSON file")
+
 	secretConfigCmd := &cobra.Command{
-		Use:   "secret <pod-name[.namespace]>",
+		Use:   "secret [<pod-name[.namespace]>]",
```

Comment:
should especially have an example since I think this is not from the standard config dump endpoint right?

Reply:
Added an example -- but it does use the same config dump URL as all of the others except the one waiting for 3362.